ElectronClient/app.js

```diff
@@ -1396,6 +1394,27 @@ class Application extends BaseApplication {
 			bridge().openDevTools();
 		}
 
+		KeymapService.setLogger(reg.logger());
+
+		const keymapPath = `${dir}/keymap-desktop.json`;
+
+		if (await fs.pathExists(keymapPath)) {
+			try {
+				let keymapFile = await fs.readFile(keymapPath, 'utf-8');
+				keymapFile = keymapFile.replace(/^\s*\/\/.*/, ''); // Strip-off comments
+
+				const customKeymap = JSON.parse(keymapFile);
+				await KeymapService.instance().setKeymap(customKeymap);
+			} catch (error) {
+				const msg = error.message ? error.message : '';
+				console.error(`Failed to parse keymap ${keymapPath}\n${msg}`);
```

Comment:
Error should be displayed to the user with an explanation on how to fix the problem. For example, by giving the line number where the error happens.

Reply:
If there's an error in the JSON formatting, now it'll be shown in an error dialogue to the user (Standard error message thrown by the `JSON.parse()` function)

I've also added more helpful error messages.

However it's a bit complicated to find the line number of some object. I think we'll have to write a mini JSON parser to do that. Is that really necessary?

Reply:
> However it's a bit complicated to find the line number of some object. I think we'll have to write a mini JSON parser to do that. Is that really necessary?

No that's fine - we just display the errors the lib gives us. If they aren't very good then so be it. Later we can indeed think about using a more advanced parser with comment support, better error message, etc. but that's later.